sfepy/discrete/fem/fields_base.py

```diff
@@ -1124,12 +1124,13 @@ def _check_region(self, region):
         """
         ok = True
         domain = region.domain
+
         if region.kind != 'cell':
             output("bad region kind! (is: %r, should be: 'cell')"
                    % region.kind)
             ok = False
 
-        elif (region.kind_tdim != domain.shape.tdim):
+        elif region.cmesh is None:
```

Comment:
How can this happen?

Reply:
`region.cmesh` points to `cmesh_tdim[3]` for a volume field. In case of shell elements, `tdim=2` and `cmesh_tdim[3]` is `None`.

Reply:
I forgot that the shell10x field inherits from `VolumeFIeld`.

`cmesh=domain.cmesh_tdim[tdim]` in Region constructor - why is that None? So tdim=3 for the shell field? Just trying to remember :)